core/src/test/java/bio/terra/pearl/core/service/workflow/EnrollmentServiceTests.java

```diff
@@ -111,23 +144,41 @@ public void testEnrollChecksConfigAllowsEnrollment(TestInfo testInfo) {
         });
     }
 
-    @Autowired
-    private EnrollmentService enrollmentService;
-    @Autowired
-    private StudyEnvironmentFactory studyEnvironmentFactory;
-    @Autowired
-    private PortalEnvironmentFactory portalEnvironmentFactory;
-    @Autowired
-    private SurveyFactory surveyFactory;
-    @Autowired
-    private StudyEnvironmentService studyEnvironmentService;
-    @Autowired
-    private StudyEnvironmentConfigService studyEnvironmentConfigService;
-    @Autowired
-    private ParticipantUserFactory participantUserFactory;
-    @Autowired
-    private StudyService studyService;
-    @Autowired
-    private PortalService portalService;
+    @Test
+    @Transactional
+    public void testEnrollProxy(TestInfo testInfo) throws JsonProcessingException {
+        PortalEnvironment portalEnv = portalEnvironmentFactory.buildPersisted(getTestName(testInfo));
+        StudyEnvironment studyEnv = studyEnvironmentFactory.buildPersisted(portalEnv, getTestName(testInfo));
+
+        EnrolleeFactory.EnrolleeAndProxy enrolleeAndProxy = enrolleeFactory.buildProxyAndGovernedEnrollee(getTestName(testInfo), portalEnv, studyEnv);
+        Enrollee proxy = enrolleeAndProxy.proxy();
+        PortalParticipantUser ppUser = portalParticipantUserService.findForEnrollee(proxy);
+        ParticipantUser user = participantUserService.find(ppUser.getParticipantUserId()).orElseThrow();
+
+        String studyShortcode = studyService.find(studyEnv.getStudyId()).get().getShortcode();
+
+        Survey survey = surveyFactory.buildPersisted(getTestName(testInfo), portalEnv.getPortalId());
+        studyEnv.setPreEnrollSurveyId(survey.getId());
+        studyEnvironmentService.update(studyEnv);
+
+        List<Answer> answers = AnswerFactory.fromMap(Map.of(
+                "qualified", true,
+                "areOver18", "yes"
+        ));
+        ParsedPreEnrollResponse response = ParsedPreEnrollResponse.builder()
+                .studyEnvironmentId(studyEnv.getId())
+                .answers(answers)
+                .qualified(true).build();
+
+        PreEnrollmentResponse savedResponse = enrollmentService.createAnonymousPreEnroll(survey.getPortalId(), studyEnv.getId(), survey.getStableId(), survey.getVersion(), response);
+
+        // now, enroll subject as proxy
+        enrollmentService.enroll(ppUser, studyEnv.getEnvironmentName(), studyShortcode, user, ppUser,
+                savedResponse.getId(), true);
+
+        Enrollee proxyAfterSubjectEnroll = enrolleeService.find(proxy.getId()).orElseThrow();
+        assertThat(proxyAfterSubjectEnroll.isSubject(), equalTo(true));
+        assertThat(proxyAfterSubjectEnroll.getPreEnrollmentResponseId(), equalTo(savedResponse.getId()));
+    }
```

Comment:
nice, seems like we could use a couple of tests (or addenda to tests above) confirming that an enrollee can't enroll twice